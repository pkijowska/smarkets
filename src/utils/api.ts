import { Events, EventDetails } from "@/types/types";

export async function getEvents(category: string): Promise<Events> {
    try {
      const response = await fetch(
        `https://api.smarkets.com/v3/events/?state=upcoming&type=${category}_match&type_domain=${category}&type_scope=single_event&with_new_type=false&sort=id&limit=20&include_hidden=false`
      );
  
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      const data = await response.json();
      return data;
    } catch (error) {
      throw new Error("Error during fetch: " + (error as Error).message);
    }
}

export const getEventDetails = async (id: string): Promise<EventDetails> => {
    try {
      const response = await fetch(
        `https://api.smarkets.com/v3/events/${id}/states/`
      );
      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }
      const data = await response.json();
      return data;
    } catch (error) {
      throw new Error("Error during fetch: " + (error as Error).message);
    }
};