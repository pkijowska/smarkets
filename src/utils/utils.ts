import { Categories } from "../types/types";

export const CATEGORIES:Categories = [
    {  label: 'Football', slug: 'football', emoji: '⚽️' },
    { label: 'Tennis', slug: 'tennis', emoji: '🎾' },
    {  label: 'Basketball', slug: 'basketball', emoji: '🏀' },
    { label: 'Baseball', slug: 'baseball', emoji: '⚾️' },
    {  label: 'Rugby Union', slug: 'rugby_union', emoji: '🏉' },
    {  label: 'Boxing', slug: 'boxing', emoji: '🥊' },
    {  label: 'Cricket', slug: 'cricket', emoji: '🏏' },
];

export function formatUTCDate(utcDateString: string) {
    const date = new Date(utcDateString);
    const day = date.getUTCDate().toString().padStart(2, '0');
    const months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
    const month = months[date.getUTCMonth()];
    const year = date.getUTCFullYear();
    const hours = date.getUTCHours().toString().padStart(2, '0');
    const minutes = date.getUTCMinutes().toString().padStart(2, '0');
    return `${day} ${month} ${year} ${hours}:${minutes} GMT`;
}