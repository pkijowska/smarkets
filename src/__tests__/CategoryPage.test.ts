import { getEvents } from "@/utils/api";
import { render, screen } from "@testing-library/react";
import CategoryPage from "@/app/events/[category]/page";

jest.mock("../utils/api", () => ({
  getEvents: jest.fn(),
  }));
  
describe("CategoryPage", () => {
  beforeEach(() => {
    (getEvents as jest.Mock).mockResolvedValue({
      events: [
        {
          bet_allowed: true,
          bettable: true,
          chart_time_period: null,
          created: '2024-02-29T08:24:00.564134Z',
          description: null,
          display_order: 0,
          end_date: null,
          full_slug: '/sport/basketball/champions-league/2024/03/19/17-30/pinar-karsiyaka-vs-hapoel-jerusalem',
          hidden: false,
          id: '43724399',
          inplay_enabled: true,
          modified: '2024-03-18T11:41:20.317371Z',
          name: 'Pinar Karsiyaka vs Hapoel Jerusalem',
          parent_id: '20310762',
          seo_description: null,
          short_name: null,
          slug: 'pinar-karsiyaka-vs-hapoel-jerusalem',
          special_rules: null,
          start_date: '2024-03-19',
          start_datetime: '2024-03-19T17:30:00Z',
          state: 'upcoming',
          type: 'basketball_match'
        }
      ]
    });
  });

  it("matches the snapshot", async () => {
    const params = { category: "Football" };
    const { container } = render(await CategoryPage({ params }));
    expect(container).toMatchSnapshot();
  })

  it("renders the name correctly", async () => {
    const params = { category: "Football" };
    const { container } = render(await CategoryPage({ params }));
    expect(container.textContent).toMatch(/Pinar Karsiyaka vs Hapoel Jerusalem/);
  });

});