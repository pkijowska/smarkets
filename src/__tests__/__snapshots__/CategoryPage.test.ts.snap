// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage matches the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card card-base"
    >
      <div
        class="innerCard"
      >
        <h3>
          Pinar Karsiyaka vs Hapoel Jerusalem
        </h3>
        <p />
        <p>
          19 Mar 2024 17:30 GMT
        </p>
      </div>
      <a
        href="/events/Football/43724399"
      >
        &gt;
         View event
      </a>
    </div>
  </div>
</div>
`;
